lib/puppet/provider/openldap_access/olc.rb

```diff
@@ -170,46 +169,46 @@ def access=(value)
 
   def self.getCountOfOlcAccess(suffix)
     countOfElement = 0
-    slapcat("(olcAccess=*)","#{getDn(suffix)}").split("\n\n").collect do |paragraph|
-      paragraph.gsub("\n ", '').split("\n").collect do |line|
+    slapcat('(olcAccess=*)', getDn(suffix).to_s).split("\n\n").map do |paragraph|
+      paragraph.gsub("\n ", '').split("\n").map do |line|
         case line
-        when /^olcAccess: /
-          countOfElement = countOfElement + 1
+        when %r{^olcAccess: }
+          countOfElement += 1
         end
       end
     end
-    return countOfElement
+    countOfElement
   end
 
   def getCurrentOlcAccess(suffix)
     i = []
-    slapcat("(olcAccess=*)","#{getDn(suffix)}").split("\n\n").collect do |paragraph|
-      paragraph.gsub("\n ", '').split("\n").collect do |line|
+    slapcat('(olcAccess=*)', getDn(suffix).to_s).split("\n\n").map do |paragraph|
+      paragraph.gsub("\n ", '').split("\n").map do |line|
         case line
-        when /^olcAccess: /
-          position, content = line.match(/^olcAccess:\s+\{(\d+)\}(.*)$/).captures
+        when %r{^olcAccess: }
+          position, content = line.match(%r{^olcAccess:\s+\{(\d+)\}(.*)$}).captures
           i << {
-            :position => position,
-            :content => content,
+            position: position,
+            content: content,
           }
         end
       end
     end
-    return i
+    i
   end
 
   def flush
-    if not @property_flush.empty?
+    unless @property_flush.empty?
       current_olcAccess = getCurrentOlcAccess(resource[:suffix])
       t = Tempfile.new('openldap_access')
       t << "dn: #{getDn(resource[:suffix])}\n"
       t << "changetype: modify\n"
       t << "replace: olcAccess\n"
-      if resource[:position]
-        position = resource[:position]
-      else
-        position = @property_hash[:position]
-      end
+      position = if resource[:position]
+                   resource[:position]
+                 else
+                   @property_hash[:position]
+                 end
```

Comment:
Isn't this the same?
```suggestion
      position = resource[:position] || @property_hash[:position]
```